### PR TITLE
[WIP] Visibility trigger for non-amp elements

### PR DESCRIPTION
**Pull request by micajuine-ho**
To be reviewed/merged after #26823's changes. Guarded with the same feature flag

For #20607

Within the IntersectionObserverPolyfill, if the element is a non-amp element (no getLayoutBox, no owner), asynchronously use `getBoundingClientRect` (offset by the `hostViewport`) to find the element's rect. 

Besides affecting VisibilityManager, this change also impacts RefreshManager (for a4a). Since it manages a4a, we're guaranteed to always get an amp-element so this should not change behavior for it.

### CODE REVIEW

File: extensions/amp-analytics/0.1/analytics-root.js [hunk 1]
```diff
@@ -275,6 +277,58 @@ export class AnalyticsRoot {
...
+   */
+  getElements_(selector) {
+    // Wait for document-ready to avoid false missed searches
+    return this.ampdoc.whenReady().then(() => {
+      const results = [];
+      const foundElements = this.getRoot().querySelectorAll(selector);
+
+      // DOM search can "look" outside the boundaries of the root, thus make
+      // sure the result is contained. Length is not supported in all browsers
+      if (foundElements && foundElements.length) {
```
Comment (jridgewell) [inline extensions/amp-analytics/0.1/analytics-root.js]: This is not necessary, `querySelector` is already guaranteed to contain everything that is returned.

What you're probably thinking of is that `querySelector('div img')` can match on `<div><root></root></div>`, but this contains check isn't going to fix that. Use `scopedQuerySelector` from `src/dom.js`

File: src/layout-rect.js [hunk 2]
```diff
@@ -280,3 +280,23 @@ export function cloneLayoutMarginsChangeDef(marginsChange) {
...
+ */
+export function getLayoutRectAsync(element, hostViewport, vsync) {
+  const rectPromise = vsync.measurePromise(() => {
+    return element./*OK*/ getBoundingClientRect();
+  });
+
+  return rectPromise.then(rect => {
+    // getBoundingClientRect() returns position relative to viewport. We want
+    // position relative to top-left of document so we add top/left of viewport
+    return hostViewport
```
Comment (jridgewell) [inline src/layout-rect.js]: Nit: couldn't you do this in the same measurer above?

Comment (micajuine-ho) [reply src/layout-rect.js]: ya, good point

**Comment by mrjoro** [comment]
Is this PR still active?

**Comment by micajuine-ho** [comment]
Yes, we are waiting for Intersection Observer experiment to be out for a while, then we will proceed with this PR.

**Comment by dvoytenko** [comment]
Just an update: intersection observer experiment is fully launched. Just cleanup is remaining. It can be now used throughout.

**Pull request closed without merging**